src/components/map/NearestAddress.tsx

```diff
@@ -0,0 +1,58 @@
+import {type CoordinatePair} from '@open-formulieren/types';
+import type React from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useAsync} from 'react-use';
+
+import {useFormSettings} from '@/hooks';
+
+import './MapAddress.scss';
+
+interface NearestAddressProps {
+  coordinates: CoordinatePair;
+}
+
+/**
+ * Retrieve and display the nearest address label for the selected coordinates.
+ */
+const NearestAddress: React.FC<NearestAddressProps> = ({coordinates}) => {
+  // XXX: this would benefit from caching, but rather than rolling our own we should
+  // probably look into useQuery/tanstack react query
+  const formSettings = useFormSettings();
+  const mapNearestLookup = formSettings?.componentParameters?.map?.mapNearestLookup;
```

Comment:
can you also prepare a (draft) PR in the SDK that provides/injects this? You can then already validate/test it via `npm run build && npm pack` + installing that tar.gz locally.